DOCS-4504 Get Started with RUM Deployment Tracking (#16064)
DataDog/documentation: content/en/real_user_monitoring/guide/setup-rum-deployment-tracking.md

---
title: Getting Started with RUM Deployment Tracking
kind: guide
beta: true
private: true
description: Learn how to set up RUM to capture new releases, track your deployments, and analyze the performance in Datadog
aliases:
- /real_user_monitoring/guide/getting-started-rum-deployment-tracking/
further_reading:
- link: '/real_user_monitoring/explorer'
  tag: 'Documentation'
  text: 'Visualize your RUM data in the RUM Explorer'
- link: "/tracing/version_tracking"
  tag: "Documentation"
  text: "Use Version tags within Datadog APM to monitor deployments"
---

## Overview
As teams iterate quickly and deploy code, it can be difficult to track down the exact change that caused a spike in errors or slower page load times. RUM Deployment Tracking will enable you to identify when a recent deployment or release is causing performance issues within your application and help you identify the source of the problem. 

## Setup

> You can use the `version` tag to monitor deployments and service behavior in support of your software deployment strategy. To get started with RUM Deployment tracking you will need to add RUM versions to your application.

Author: the important part of deployment tracking is that you have the RUM version

> You can use the `version` tag to monitor deployments and service behavior in support of your software deployment strategy. To get started with RUM Deployment tracking you will need to add RUM versions to your application.

Reviewer: suggestion modified

datadogRum.init({
  ...
  version: '1.0.0',
  ...
});
```

{{% /tab %}}
{{% tab "CDN async" %}}

```html
<script>
  (function(h,o,u,n,d) {
     h=h[d]=h[d]||{q:[],onReady:function(c){h.q.push(c)}}
     d=o.createElement(u);d.async=1;d.src=n
     n=o.getElementsByTagName(u)[0];n.parentNode.insertBefore(d,n)
  })(window,document,'script','https://www.datadoghq-browser-agent.com/datadog-rum-v4.js','DD_RUM')
  DD_RUM.onReady(function() {
    DD_RUM.init({
      ...
      version: '1.0.0',
      ...
    })
  })
</script>
```

{{% /tab %}}
{{% tab "CDN sync" %}}

```html
<script src="https://www.datadoghq-browser-agent.com/datadog-rum-v4.js" type="text/javascript"></script>
<script>
  window.DD_RUM &&
    window.DD_RUM.init({
      ...
      version: '1.0.0',
      ...
    })
</script>
```
{{% /tab %}}
{{< /tabs >}}

### Mobile RUM

#### Android RUM

The version tag is captured automatically from the application’s manifest.

#### iOS RUM

The version tag is captured automatically from the application’s app’s `info.plist`.

## Analyze your deployment performance in RUM

{{< tabs >}}
{{% tab "Browser RUM" %}}

### Using version tags on the Application Overview page

> An application configured with version tags has a Deployment Tracking section on the Application Overview page. The Deployment Tracking section shows all versions of the application and services that were active during the selected time interval. 

Reviewer: ```suggestion
An application configured with version tags has a **Deployment Tracking** section on the Application Overview page. The **Deployment Tracking** section shows all versions of the application and services that were active during the selected time interval. 
```

> This enables you to quickly roll back release candidates as soon as you spot an issue so you can avoid negative user experiences. These out of the box graphs are aggregated across versions, making it easier to identify problems in your application before they turn into serious issues.

Reviewer: ```suggestion
This enables you to roll back release candidates as soon as you notice an issue so you can avoid negative user experiences. These out of the box graphs are aggregated across versions, making it easier to identify problems in your application before they turn into serious issues.
```

You will see:
- P75 Loading Time by Version
- Total User Sessions by Version
- Error Rate by Version

In the table below these widgets, you will see:
- The version names deployed for the application and its services over the timeframe.
- Number of user session for that version
- Average errors per view
- P75 Loading Time

> - P75 for Core Web Vitals

Reviewer: What are we trying to achieve with these bulleted lists? I don't think they are helpful for the user as-is, if they're just stating what's there, they feel more like marketing end less like technical documentation.

Author: Yes, this is just stating what's there. I was following a similar pattern to the [APM deployment tracking page ](https://docs.datadoghq.com/tracing/services/deployment_tracking/)but am not opposed to removing this.

Reviewer: It looks like on that page they're framed more as "these are the available columns" - I think we could maybe adjust it to be the same if that's what the functionality is, instead of "you will see" as long as is it something that they can add or remove, or are they always there?

Author: yes, it's the same functionality as APM. they are always there, users will need to duplicate/copy the widgets to a dashboard to modify them

Author: Should we leave this as is then? Or is there some wording change we should make?

These widgets can be exported to dashboards and monitors.

{{< img src="real_user_monitoring/guide/setup-rum-deployment-tracking/browser-rum-app-overview-deployment-tracking.png" alt="Browser Deployment Tracking in RUM Application Overview" style="width:100%;">}}

### Deployment Comparison

> Click on any version row in the List of Versions table to open a version comparison page, allowing you to compare two versions of the same service. By default, the selected version will be compared to all previous versions but you can change it to compare any two versions within the past 30 days.

Reviewer: ```suggestion
Click on any version row in the **List of Versions*** table to open a version comparison page, allowing you to compare two versions of the same service. By default, the selected version is compared to all previous versions. You can change the selection to compare any two versions within the past 30 days.
```

> Similar to the graphs on the Application Overview page, User Sessions, Core Web Vitals, and Errors graphs show an overview of a deployment rollout or spikes in error rates. On this page, the graphs highlight the selected versions for comparison and leave all other versions in gray for additional context.

Reviewer: ```suggestion
Similar to the graphs on the **Application Overview** page, the **User Sessions**, **Core Web Vitals**, and **Errors** graphs show an overview of a deployment rollout or spikes in error rates. On this page, the graphs highlight the selected versions for comparison and display all other versions in gray for additional context.
```

> As you are monitoring your release, this makes it easy for you to compare the performance of code deployments against existing live code to verify that new code is performing properly and that no new errors have surfaced in between versions. 

Reviewer: ```suggestion
As you are monitoring your release, this helps you to compare the performance of code deployments against existing live code to verify that new code is performing properly, and that no new errors have surfaced in between versions. 
```

{{< img src="real_user_monitoring/guide/setup-rum-deployment-tracking/browser-deployment-tracking-comparison.png" alt="Browser Deployment Tracking Comparison" style="width:75%;">}}

> The Issues tab, lists differences in errors detected for each the two versions, highlighting:

Reviewer: ```suggestion
The **Issues** tab lists differences in errors detected for each of the two versions, highlighting:
```

- Error Count by Version 
- % of Views with Errors by Version 
- Error Tracking issues

{{< img src="real_user_monitoring/guide/setup-rum-deployment-tracking/browser-deployment-tracking-comparison-error.png" alt="Browser Deployment Tracking Comparison Errors" style="width:75%;">}}

> ### Explore the RUM Deployment Tracking Powerpacks

Reviewer: ```suggestion
### Explore the RUM deployment tracking powerpacks
```

> You can quickly add deployment tracking for your RUM services to dashboards using the powerpacks menu on a dashboard and searching for the “Deployment Version Tracking” powerpack. You can then iterate and add any other widget to your dashboards to help your teams release new features safely.

Reviewer: ```suggestion
You can add deployment tracking for your RUM services to dashboards using the powerpacks menu on a dashboard and searching for the “Deployment Version Tracking” powerpack. You can then iterate and add any other widgets to your dashboards to help your teams release new features safely.
```

{{< img src="real_user_monitoring/guide/setup-rum-deployment-tracking/browser-deployment-tracking-powerpack.png" alt="Browser Deployment Tracking Powerpack" style="width:75%;">}}

{{% /tab %}}
{{% tab "Mobile RUM" %}}

### Using version tags on the Application Overview page

> An application configured with version tags has a Deployment Tracking section on the Application Overview page. The Deployment Tracking section shows all versions of the application and services that were active during the selected time interval. 

Reviewer: ```suggestion
An application configured with version tags has a **Deployment Tracking** section on the Application Overview page. The **Deployment Tracking** section shows all versions of the application and services that were active during the selected time interval. 
```

> This enables you to adopt modern deployment strategies with peace of mind by letting you quickly roll back release candidates as soon as you spot an issue so you can avoid negative user experiences. These out of the box graphs are aggregated across versions, making it easy to spot problems in your application before they turn into serious issues.

Reviewer: ```suggestion
This enables you to quickly roll back release candidates as soon as you spot an issue so you can avoid negative user experiences. These out of the box graphs are aggregated across versions, making it easier to identify problems in your application before they turn into serious issues.
```

> This enables you to adopt modern deployment strategies with peace of mind by letting you quickly roll back release candidates as soon as you spot an issue so you can avoid negative user experiences. These out of the box graphs are aggregated across versions, making it easy to spot problems in your application before they turn into serious issues.

Reviewer: We should add the tabs after this section, since they're identical and the content is being duplicated. Duplication like this adds a lot of overhead for us over the life of the doc.

Author: the reason that i added the tabs before this is that the entire section has some variability for web vs. mobile apps. there are some sections that are duplicated but we would be adding a much of tabs throughout the sections. Wasn't sure if that made sense from a readability standpoint

You will see:
- Average Application Start Time by Version
- Total User Sessions by Version
- Error Rate by Version

In the table below these widgets, you will see:
- The version names deployed for the application and its services over the timeframe.
- Number of app launches for that version
- Error Rate
- Crash Rate
- P90 Application Start Time

> 

Reviewer: Same discussion as above for this section, either remove or reframe it as "These are the available columns" ?

These widgets can be exported to dashboards and monitors.

{{< img src="real_user_monitoring/guide/setup-rum-deployment-tracking/mobile-rum-app-overview-deployment-tracking.png" alt="Mobile Deployment Tracking in RUM Application Overview" style="width:100%;">}}

> ### Deployment Comparison

Reviewer: ```suggestion
### Deployment comparison
```

> Click on any version row in the List of Versions table to open a version comparison page, allowing you to compare two versions of the same service. By default, the selected version will be compared to all previous versions but you can change it to compare any two versions within the past 30 days.

Reviewer: ```suggestion
Click on any version row in the **List of Versions** table to open a version comparison page, allowing you to compare two versions of the same service. By default, the selected version is compared to all previous versions. You can change the selection to compare any two versions within the past 30 days.
```

> Similar to the graphs on the Application Overview page, User Sessions, Mobile Vitals, and Errors graphs show an overview of a deployment rollout or spikes in error rates. On this page, the graphs highlight the selected versions for comparison and leave all other versions in gray for additional context.

Reviewer: ```suggestion
Similar to the graphs on the **Application Overview** page, the **User Sessions**, **Core Web Vitals**, and **Errors** graphs show an overview of a deployment rollout or spikes in error rates. On this page, the graphs highlight the selected versions for comparison and display all other versions in gray for additional context.
```

Author: I think this should be mobile vitals? not core web vitals

Reviewer: suggestion updated

> As you are monitoring your release, this makes it easy for you to compare the performance of code deployments against existing live code to verify that new code is performing properly and that no new errors have surfaced in between versions. 

Reviewer: Again, this duplicated content makes it hard to maintain as well as review consistently

Author: It is also not exactly duplicated as we mention Mobile Vitals vs Web vitals

Reviewer: Ok, for now let's go with the tabs as-is, to your point I do think it makes more sense to have less of them than a ton sprinkled throughout. Hopefully in the future though we can figure out a happy medium for the content to be re-used and easier to translate and edit without duplication.

{{< img src="real_user_monitoring/guide/setup-rum-deployment-tracking/mobile-deployment-tracking-comparison.png" alt="Mobile Deployment Tracking Comparison" style="width:75%;">}}

> The Issues tab, lists differences in errors detected for each the two versions, highlighting:

Reviewer: ```suggestion
The **Issues** tab lists differences in errors detected for each of the two versions, highlighting:
```

- Error Count by Version 
- % of Views with Errors by Version 
- Error Tracking issues

{{< img src="real_user_monitoring/guide/setup-rum-deployment-tracking/mobile-deployment-tracking-comparison-error.png" alt="Mobile Deployment Tracking Comparison Errors" style="width:75%;">}}

### Explore the RUM Deployment Tracking Powerpacks

> You can quickly add deployment tracking for your RUM services to dashboards using the powerpacks menu on a dashboard and searching for the “Deployment Version Tracking” powerpack. You can then iterate and add any other widget to your dashboards to help your teams release new features safely.

Reviewer: ```suggestion
You can add deployment tracking for your RUM services to dashboards using the powerpacks menu on a dashboard and searching for the “Deployment Version Tracking” powerpack. You can then iterate and add any other widgets to your dashboards to help your teams release new features safely.
```
